client/src/graphics/window.rs

```diff
@@ -707,3 +673,32 @@ struct Frame {
     /// Semaphore used to ensure the frame isn't presented until rendering completes
     present: vk::Semaphore,
 }
+
+#[derive(Default)]
+struct InputState {
+    forward: bool,
+    back: bool,
+    left: bool,
+    right: bool,
+    up: bool,
+    down: bool,
+    jump: bool,
+    clockwise: bool,
+    anticlockwise: bool,
+    last_frame: Option<Instant>,
```

Comment:
Out of all fields, this one seems the most out of place in the `InputState` struct, but at the same time, it does provide context for the rest of the fields, and in the context of a simulation, "amount of time passed" is indeed one of the inputs, so I don't think moving this field somewhere else would necessarily be any better.

Reply:
Eh, lifted it out.